apps/src/lib/kits/maker/ui/SetupChecklist.jsx

```diff
@@ -225,27 +225,23 @@ export default class SetupChecklist extends Component {
   }
 
   installFirmwareSketch() {
-    let firmataFromBoardType;
-    switch (this.state.boardTypeDetected) {
-      case BOARD_TYPE.EXPRESS:
-        firmataFromBoardType =
-          'https://learn.adafruit.com/adafruit-circuit-playground-express/code-org-csd';
-        break;
-      case BOARD_TYPE.MICROBIT:
-        firmataFromBoardType =
-          'https://github.com/microbit-foundation/microbit-firmata#installing-firmata-on-your-bbc-microbit';
-        break;
-      default:
-        firmataFromBoardType =
-          'https://learn.adafruit.com/circuit-playground-firmata/overview';
+    let firmataMarkdown;
+    if (this.state.boardTypeDetected === BOARD_TYPE.MICROBIT) {
+      firmataMarkdown = applabI18n.makerSetupInstallFirmataMB({
+        firmataURL:
+          'https://github.com/microbit-foundation/microbit-firmata#installing-firmata-on-your-bbc-microbit'
+      });
+    } else {
```

Comment:
is it safe to assume the user is trying to connect a CP if `boardTypeDetected` isn't micro:bit? i might be missing a piece of info, but i think we could have something similar that covers all our bases (sorry for the pseudocode, but it's probably clearer than me trying to explain):

```javascript
if (microbit) {
  // return micro:bit message
} else if (circuitPlayground) {
  // return CP message
} else {
  // return both messages, maybe as something like:
  //
  // Sorry, we can't tell what board you're trying to connect! Depending on your board type, you should...
  //   - [CP message]
  //   - [micro:bit message]
}

```

Reply:
`is it safe to assume the user is trying to connect a CP if boardTypeDetected isn't micro:bit?` - At this point in the code, we know if we are a CP or MB board. Step 2 is where we detect which board we are using. Step 3 is where we create an instance of the board (https://github.com/code-dot-org/code-dot-org/blob/30f8a7454bae694a78b63ac580bfc86faadb9bea/apps/src/lib/kits/maker/util/SetupChecker.js#L55) depending on the type of board that is selected. Step 4 (where this edit is) is where we ping the board to check if we can reach the components (i.e. is the right firmware installed?). At this point in the process, we have created either a MB or CP board instance (or have failed doing so). Since the board instance for CPC/CPE is the same, this is the first time we try to identify which version of the CP we are using. When we have an unidentified board, we try to make a CircuitPlayground board to see how far that takes us. (https://github.com/code-dot-org/code-dot-org/blob/30f8a7454bae694a78b63ac580bfc86faadb9bea/apps/src/lib/kits/maker/util/SetupChecker.js#L55) That being said, I think having the three-way conditional is a good way to cover our bases.

Reply:
Wait, I just remembered that we also use this error message if Step 3 fails which reinforces that the three-part conditional is a good idea here.